backend/ibutsu_server/widgets/jenkins_heatmap.py

```diff
@@ -30,106 +33,94 @@ def _calculate_slope(x_data):
 
 
 def _get_builds(job_name, builds, project=None):
-    """ Gets the number of unique builds in the DB """
-    aggregation = [
-        {
-            "$match": {
-                "metadata.jenkins.job_name": job_name,
-                "metadata.jenkins.build_number": {"$exists": True},
-            }
-        },
-        {"$sort": {"start_time": DESCENDING}},
-        {"$limit": HEATMAP_RUN_LIMIT},
-        {
-            "$group": {
-                "_id": "$metadata.jenkins.build_number",
-                "start_time": {"$min": "$start_time"},
-            }
-        },
-        {"$sort": {"start_time": DESCENDING}},
-        {"$limit": builds},
-    ]
+    filters = [f"metadata.jenkins.job_name={job_name}", "metadata.jenkins.build_number@y"]
     if project:
-        aggregation[0]["$match"].update({"metadata.project": project})
-    builds = list(mongo.runs.aggregate(aggregation))
-    return [build["_id"] for build in builds]
+        filters.append(f"metadata.project={project}")
+
+    # generate the group_field
+    group_field = string_to_column("metadata.jenkins.build_number", Run)
+
+    # create the query
+    query = (
+        session.query(group_field.label("build_number"))
+        .group_by("build_number")
+        .order_by(desc("build_number"))
+    )
+
+    # add filters to the query
+    query = apply_filters(query, filters, Run)
+
+    # make the query
+    query_data = query.limit(builds).all()
+
+    return [build_number[0] for build_number in query_data]
```

Comment:
You can actually drop the `.all()` and iterate on the resultant object of `limit()` (it's actually just another `Query` object). The query is iterable, and will do smart iteration against the database.